src/api/services/ProposalActionService.ts

```diff
@@ -104,44 +110,92 @@ export class ProposalActionService {
-                const proposalResult = await this.createProposalResult(createdProposal)
-                    .catch();
-
-                // TODO: Validation??
-                // - sanity check for proposal start/end blocks vs current one
-
-                // if Proposal is of type ITEM_VOTE, and if ListingItem exists, link them
-                if (createdProposal.type === ProposalType.ITEM_VOTE) {
-                    await this.listingItemService.findOneByHash(createdProposal.title)
-                        .then( async listingItemModel => {
-                            const listingItem = listingItemModel.toJSON();
-                            await this.listingItemService.updateProposalRelation(listingItem.id, createdProposal.hash);
-
-                            // get the market and vote
-                            await this.marketService.findByAddress(marketplaceMessage.market || '')
-                                .then(async marketModel => {
-                                    const market = marketModel.toJSON();
-                                    await this.voteForListingItemProposal(createdProposal, market);
-                                });
-                        })
-                        .catch(reason => {
-                            this.log.warn('received Proposal, but theres no ListingItem for it yet...');
-                        });
-                }
-
-                // createdProposalModel = await this.proposalService.findOne(createdProposal.id);
-                // return createdProposalModel.toJSON();
-                return SmsgMessageStatus.PROCESSED;
-            })
-            .catch(reason => {
-                return SmsgMessageStatus.WAITING;
-            });
+        const currentBlock = await this.coreRpcService.getBlockCount();
+        // TODO: Validation??
+        // - sanity check for proposal start/end blocks vs current one
+
+        let proposal: resources.Proposal;
+        let proposalResult;
+        let tmpProposal;
+        try {
+          tmpProposal = await this.proposalService.findOneByItemHash(proposalCreateRequest.item);
+        } catch (ex) {
+          // Do nothing
+        }
+        if (tmpProposal) {
+          proposal = tmpProposal.toJSON();
+          // We have a proposal for this item already.
+          if (proposal.createdAt < smsgMessage.createdAt) {
+            try {
+              // This is an older proposal, but for some reaason we received it *after*.
+              // Replace the existing proposal.
+              proposalResult = await this.proposalResultService.findOneByProposalHash(proposal.hash);
+              tmpProposal = await this.proposalService.update(proposal.id, proposalCreateRequest);
+              proposal = tmpProposal.toJSON();
+              proposalResult = await this.createProposalResult(proposal);
+            } catch (ex) {
+              this.log.warn(ex);
+              return SmsgMessageStatus.WAITING;
```

Comment:
WAITING should be used for smsgMessages that are waiting for some other message to arrive and which will then be processed again.. sorry there's no documentation for this :)